packages/client-generator-ts/src/TSClient/file-generators/CommonInputTypesFile.ts

```diff
@@ -0,0 +1,49 @@
+import * as ts from '@prisma/ts-builders'
+
+import { GenerateContext } from '../GenerateContext'
+import { InputType } from '../Input'
+
+const jsDocHeader = `/**
+ * This file exports various common sort, input & filter types that are not directly linked to a particular model.
+ *
+ * 🟢 You can import this file directly.
+ */
+`
+
+export function createCommonInputTypeFiles(context: GenerateContext) {
+  const imports = [
+    ts.moduleImport(context.runtimeImport).asNamespace('runtime'),
+    ts.moduleImport(context.importFileName(`./enums`)).asNamespace('$Enums'),
+    ts.moduleImport(context.importFileName(`./internal/prismaNamespace`)).asNamespace('Prisma').typeOnly(),
+  ].map((i) => ts.stringify(i))
+
+  const genericInputTypes = context.dmmf.inputObjectTypes.prisma
+    // Only contains generic input types that are not directly linked to a particular model
+    .filter((i) => !i.meta?.grouping)
+    .map((inputType) => {
+      if (inputType.name.includes('Json') && inputType.name.includes('Filter')) {
+        const needsGeneric = context.genericArgsInfo.typeNeedsGenericModelArg(inputType)
+        const innerName = needsGeneric ? `${inputType.name}Base<$PrismaModel>` : `${inputType.name}Base`
+        const typeName = needsGeneric ? `${inputType.name}<$PrismaModel = never>` : inputType.name
+        // This generates types for JsonFilter to prevent the usage of 'path' without another parameter
+        const baseName = `Required<${innerName}>`
+        return `export type ${typeName} = 
+| Prisma.PatchUndefined<
+    Prisma.Either<${baseName}, Exclude<keyof ${baseName}, 'path'>>,
+    ${baseName}
+  >
+| Prisma.OptionalFlat<Omit<${baseName}, 'path'>>
```

Comment:
This seems to be the same as in `ModelFile.ts`, should we extract this to a function?

Reply:
Correct. Originally thought those two might evolve differently over my refactoring. Not the case though. Moved this logic into the `InputType` class. I checked other usages of this class and seems to be safe to have that codepath in there always.